clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -84,8 +85,24 @@ public void close() {
      *
      * @param duration  The duration to use for the wait time.
      * @param unit      The time unit to use for the wait time.
+     * @deprecated Since 2.1. Use {@link #close(Duration)} or {@link #close()}.
      */
-    public abstract void close(long duration, TimeUnit unit);
+    @Deprecated
+    public void close(long duration, TimeUnit unit) {
```

Comment:
I wonder if a safer way to do this from a compatibility perspective would be to provide a default method for `close(Duration)` which invokes `close(long, TimeUnit)`. Similarly for the producer.

Reply:
Got it. However, it will be hard to remove deprecated methods if the compatibility guarantee includes the "inheritance"...

Reply:
I think deprecation covers this. Eventually at a major release version, we can drop the old methods. But it's reasonable to expect that we don't break compatibility until the deprecation window has closed.

Reply:
> I think deprecation covers this. Eventually at a major release version, we can drop the old methods. But it's reasonable to expect that we don't break compatibility until the deprecation window has closed.

Thanks for the explanation!